egs2/TEMPLATE/speechlm1/speechlm.sh

```diff
@@ -0,0 +1,734 @@
+                # for discrete operations, we will also generate a vocabulary.
+
+                if [ ! -f ${data_audio}/${dset}/${_name} ]; then
+                    echo "File ${data_audio}/${dset}/${_name} is missing. Exit" || exit 1;
+                fi
+
+                if [ ${_modality} == "ssl" ]; then
+                    echo "do ssl tokenization" && exit 1;
+
+                elif [ ${_modality} == "codec" ]; then
+                    echo "Codec Tokenization: ${data_audio}/${dset}/${_name} -> ${data_feats}/${dset}/${_name}"
+                    scripts/feats/codec_tokenization.sh \
+                        --src_dir ${data_audio}/${dset} --tgt_dir ${data_feats}/${dset} \
+                        --codec_fs ${fs} \
+                        --file_name ${_name} --nj ${nj} --codec_choice ${codec_choice} ${codec_opts}
+
+                elif [ ${_modality} == "g2p" ]; then
+                    echo "Find G2P vocabulary and copy text"
+                    ${python} -m espnet2.bin.tokenize_text \
+                        --token_type "phn" -f 2- \
+                        --input "${data_audio}/${dset}/${_name}" \
+                        --output "${data_feats}/${dset}/token_lists/g2p_token_list" \
+                        --non_linguistic_symbols "${nlsyms_txt}" \
+                        --cleaner "${cleaner}" \
+                        --g2p "${g2p}" \
+                        --write_vocabulary true
+                    cp "${data_audio}/${dset}/${_name}" "${data_feats}/${dset}/${_name}"
+                
+                elif [ ${_modality} == "spk" ]; then
+                    echo "copy utt2spk file"
+                    cp "${data_audio}/${dset}/${_name}" "${data_feats}/${dset}/${_name}"
+
+                else
+                    echo "Unsupported modality ${_modality}" && exit 1;
+                fi
+
+                opts+="--file_modality_type ${data_feats}/${dset}/${_name},${_modality},${_type} "
+                if [ -f ${data_feats}/${dset}/token_lists/${_modality}_token_list ]; then
+                    opts+="--token_list ${data_feats}/${dset}/token_lists/${_modality}_token_list "
+                fi
+            done
+
+            # The metadata for this dataset/task is saved in a yaml file
+            ${python} pyscripts/utils/make_speechlm_json.py \
+                --task ${task} \
+                --output_json ${data_feats}/${dset}/data.json \
+                ${opts}
+        done
+    fi
+
+else
+    log "Skip the stages for data preparation"
+fi
+
+# ========================== Data preparation is done here. ==========================
+
+
+if ! "${skip_train}"; then
+
+    _data_opts=""
+    if [ -z ${train_jsons} ]; then
+        train_jsons=${data_feats}/${train_set}/data.json
+        log "No train_jsons provided. Use the prepared one: ${train_jsons}"
+    fi
+    
+    for train_json in $train_jsons; do
+        _data_opts+="--train_data_path_and_name_and_type ${train_json},json,json "
```

Comment:
Is name "json" a good indicator of the information (would recommend a better name?)

Reply:
"json,json" may not be a good option. 
The first "json" is the name but actually will not be in use. We can change it to task/dataset name maybe.
The second should be "json" as it specify this is a json file that specifies a dataset.

Reply:
Agree, maybe change it to task instead.